interfaces/builtin/repowerd.go

```diff
@@ -0,0 +1,335 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var repowerdPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the repowerd service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# HW alarms
+/dev/alarm rw,
+
+# Suspend
+/sys/power/** rw,
+
+# Backlight
+/sys/**/brightness rw,
```

Comment:
I did the survey, but unfortunately I am not able to restrict the rule further. Although repowerd reads backlight info from subdirs of /sys/class/backlight/ (and /sys/class/leds/lcd-backlight on some devices) these subdirs are actually links to arbitrary paths within /sys/... , depending on device, graphics card etc. Using a /sys/class/backlight/... rule doesn't work since apparmor also checks permissions for the linked directories, not just the links themselves.